plugin-svn/che-plugin-svn-ext-subversion/src/main/java/org/eclipse/che/ide/ext/svn/server/SubversionConfigurationChecker.java

```diff
@@ -0,0 +1,189 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.svn.server;
+
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Predicate;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Checks Subversion configuration to ensure properly work SVN-extension in Codenvy.
+ * <p/>
+ * Here we check that we have correct setting for svn ignore.
+ * Must be ignored files:
+ * <pre>.codenvy
+ *  .vfs</pre>
+ * like system files for Codenvy.
+ *
+ * @author Vladyslav Zhukovskyi
+ */
+@Singleton
+public class SubversionConfigurationChecker {
+
+    private static final Set<String> SUBVERSION_IGNORE_PATTERNS = new LinkedHashSet<>();
+
+    static {
+        SUBVERSION_IGNORE_PATTERNS.add(".codenvy");
+        SUBVERSION_IGNORE_PATTERNS.add(".vfs");
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(SubversionConfigurationChecker.class);
+
+    private final Path GLOBAL_SUBVERSION_CONFIG_FILE_PATH;
+
+    protected SubversionConfigurationChecker() {
+        GLOBAL_SUBVERSION_CONFIG_FILE_PATH = Paths.get(System.getProperty("user.home") + "/.subversion/config");
+    }
+
+    // Constructor need for unit-tests
+    protected SubversionConfigurationChecker(Path subversionConfigFile) {
+        GLOBAL_SUBVERSION_CONFIG_FILE_PATH = subversionConfigFile;
+    }
+
+    @PostConstruct
+    public void start() {
+        try {
+            ensureExistingSvnConfigFile();
+            checkAndUpdateConfigFile();
+        } catch (IOException e) {
+            LOG.error(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Return loaded configuration file.
+     *
+     * @return configuration file
+     */
+    protected Path getLoadedConfigFile() {
+        return GLOBAL_SUBVERSION_CONFIG_FILE_PATH;
+    }
+
+    /**
+     * Checks if Subversion configuration file exist and if none, then tries to create it.
+     *
+     * @throws IOException
+     *         if creation was failed
+     */
+    protected void ensureExistingSvnConfigFile() throws IOException {
+        if (Files.notExists(GLOBAL_SUBVERSION_CONFIG_FILE_PATH)) {
+            Files.createFile(GLOBAL_SUBVERSION_CONFIG_FILE_PATH);
+        }
+    }
+
+    /**
+     * Checks existing config file for global-ignore property filling and concatenate system default values need to proper work of Codenvy.
```

Comment:
Added little descriptions which way is proper for Codenvy works. Here is little list of which directories should be ignored on user's level.